### PR TITLE
fix(parsers.json_v2): Check if gpath exists and support optional in fields/tags

### PR DESCRIPTION
**Pull request by sspaink**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolve: #10670
resolve: #10776

https://github.com/influxdata/telegraf/pull/10468 introduced a bug by checking if a gpath exists by comparing the return type to `null`, which prevented `null` being allowed in a JSON file. Instead it should have used the provided `Exists()` method as suggested in  #10670 by @kitlaan and @kgrimsby thanks! :tada:. This PR updates the parser to use `Exists()` when appropriate. It was also suggested to error only after all the paths have been checked which seems like a good idea, but I am hesitant introducing that into this fix because it does change behavior and maybe keeping this PR just to fix the bug?

https://github.com/influxdata/telegraf/pull/10468 introduced the configuration option `optional` so that the json_v2 parser can work better with the mqtt_consumer plugin when the paths aren't guaranteed in the JSON. This PR overlooked the global field and tag to include the configuration as well, this was pointed out in #10776. This PR allows setting `optional` to field and tag as well.

### CODE REVIEW

File: plugins/parsers/json_v2/README.md [hunk 1]
```diff
@@ -99,13 +100,15 @@ Using this field configuration you can gather a non-array/non-object values. Not
 * **path (REQUIRED)**: A string with valid GJSON path syntax to a non-array/non-object value
 * **name (OPTIONAL)**: You can define a string value to set the field name. If not defined it will use the trailing word from the provided query.
 * **type (OPTIONAL)**: You can define a string value to set the desired type (float, int, uint, string, bool). If not defined it won't enforce a type and default to using the original type defined in the JSON (bool, float, or string).
+* **opitional (OPTIONAL)**: Setting optional to true will suppress errors if the configured Path doesn't match the JSON. This should be used with caution because it removes the safety net of verifying the provided path. This was introduced to support situations when parsing multiple incoming JSON payloads with wildcards. ([More context](https://github.com/influxdata/telegraf/issues/10072))
```
Comment (kitlaan) [inline plugins/parsers/json_v2/README.md:103]: Typo `opitional`

Comment (sspaink) [reply plugins/parsers/json_v2/README.md]: good catch, thanks

**Review by kitlaan — APPROVED** [review]

File: plugins/parsers/json_v2/testdata/wrong_path/telegraf.conf [hunk 2]
```diff
@@ -1,12 +1,5 @@
 # Example taken from: https://github.com/influxdata/telegraf/issues/7097
 
-[[inputs.file]]
```
Comment (reimda) [inline plugins/parsers/json_v2/testdata/wrong_path/telegraf.conf:3]: Was this removed intentionally?

File: plugins/parsers/json_v2/testdata/wrong_path/telegraf.conf [hunk 3]
```diff
@@ -15,32 +8,3 @@
             path = "correct"
             [[inputs.file.json_v2.object.tag]]
                 path = "wrong"
-
```
Comment (reimda) [inline plugins/parsers/json_v2/testdata/wrong_path/telegraf.conf:18]: Was this removed intentionally?

Comment (sspaink) [reply plugins/parsers/json_v2/testdata/wrong_path/telegraf.conf]: It wasn't, good catch!

File: plugins/parsers/json_v2/testdata/wrong_path/telegraf.conf (hunk 2 above)
Comment (sspaink) [reply plugins/parsers/json_v2/testdata/wrong_path/telegraf.conf]: It wasn't, good catch!

**Review by reimda — APPROVED** [review]

**Pull request merged**